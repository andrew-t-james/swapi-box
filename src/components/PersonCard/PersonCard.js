import React from 'react';
import Button from '../Button/Button';
import PropTypes from 'prop-types';

const PersonCard = props => {
  const { name, species, homeworld, population, favorite, updateFavorites } = props;
  const person = {
    name,
    species,
    homeworld,
    population,
    favorite: true
  };

  return (
    <div>
      <h2>{name}</h2>
      <p>{species}</p>
      <p>{homeworld}</p>
      <p>{population}</p>
      <span className="person-card__favorite" onClick={() => updateFavorites(person)}>
        <Button name="⭐" />
      </span>
    </div>
  );
};

PersonCard.propTypes = {
  name: PropTypes.string,
  species: PropTypes.string,
  favorite: PropTypes.bool,
  homeworld: PropTypes.string,
  population: PropTypes.string,
  updateFavorites: PropTypes.func
};

export default PersonCard;
