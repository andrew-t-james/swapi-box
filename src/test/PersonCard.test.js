import React from 'react';
import { shallow } from 'enzyme';
import PersonCard from '../components/PersonCard/PersonCard';

describe('PersonCard', () => {
  let wrapper;
  const mockFunc = jest.fn();
  const mockProps = {
    name: 'Wookie'
  };

  beforeEach(() => wrapper = shallow(
    <PersonCard
      props={{mockProps}}
      updateFavorites={mockFunc}
    />
  ));

  afterEach(() => wrapper.unmount());

  test('should call updateFavorites onClick', () => {
    wrapper.find('.card__favorite').first().simulate('click');
    expect(mockFunc).toHaveBeenCalled();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
