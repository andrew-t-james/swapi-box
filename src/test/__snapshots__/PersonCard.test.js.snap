// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonCard
    props={
        Object {
            "mockProps": Object {
              "name": "Wookie",
            },
          }
    }
    updateFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <p />,
        <p />,
        <p />,
        <span
          className="person-card__favorite"
          onClick={[Function]}
>
          <Button
                    name="⭐"
          />
</span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            name="⭐"
/>,
          "className": "person-card__favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "⭐",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <p />,
          <p />,
          <p />,
          <span
            className="person-card__favorite"
            onClick={[Function]}
>
            <Button
                        name="⭐"
            />
</span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              name="⭐"
/>,
            "className": "person-card__favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "⭐",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
